lib/MusicBrainz/Server/Data/Editor.pm

```diff
@@ -194,7 +194,7 @@ sub search_by_email {
 
     my $query = 'SELECT ' . $self->_columns .
         ' FROM ' . $self->_table .
-        q" WHERE regexp_replace(regexp_replace(email, '(\+.*@)', '@'), '\.(.*@)', '\1', 'g') ~ ?" .
```

Comment:
Just to make sure, does it run? I see you removed the leading space, but `$self->_table` is just `'editor'`, so I'd think that'd result in broken syntax (`editorWHERE`).

Reply:
Good catch. Fixed it and added tests in <https://github.com/metabrainz/musicbrainz-server/pull/2102/commits/b19fc87e0475d2af29e79e0ebfffaf58ef6eb135>.

I got distracted by the uneven indentation of SQL keywords. Maybe we could use [indented here-documents](https://perldoc.perl.org/perlop#Indented-Here-docs) `<<~EOSQL` to embed SQL code? That would just require a couple of extra variables to replace `->` attribute/function calls.

Reply:
I've used regular here-docs a lot in the past (grep `EOSQL`) and agree stripped ones would be preferable. :) The only reason I hadn't used those is that [they were added in Perl 5.26](https://www.effectiveperlprogramming.com/2016/12/strip-leading-spaces-from-here-docs-with-v5-26/) AFAICT.